### ISSUE TITLE
ValueError: Expected x_min for bbox (-0.001953125, 0.18359375, 0.158203125, 0.640625, 0) to be in the range [0.0, 1.0], got -0.001953125.

### ISSUE THREAD

**Issue by Ycrazyprimitive** [open-mmlab/mmdetection#5657]
**Describe the bug**
When I use ‘Albu’ for data enhancement, the following bug occurs：
ValueError: Expected x_min for bbox (-0.001953125, 0.18359375, 0.158203125, 0.640625, 0) to be in the range [0.0, 1.0], got -0.001953125.

**Reproduction**
My Albu‘s configs：
albu_train_transforms = [
    dict(
        type='HorizontalFlip',
        p=0.5
    ),
    dict(
        type='VerticalFlip',
        p=0.5
    )
]
    dict(type='Albu',
         transforms=albu_train_transforms,
         bbox_params=dict(
             type='BboxParams',
             format='pascal_voc',
             label_fields=['gt_labels'],
             min_visibility=0.1,
             min_area=0.1,
             filter_lost_elements=True),
         keymap={
             'img': 'image',
             'gt_bboxes': 'bboxes'
         },
         update_pad_shape=False,
         skip_img_without_anno=True), 

My dataset is a SAR image dataset in VOC format.

**Environment**

1. environment information:
sys.platform: linux
Python: 3.7.10 | packaged by conda-forge | (default, Feb 19 2021, 16:07:37) [GCC 9.3.0]
CUDA available: True
GPU 0: GeForce RTX 3090
CUDA_HOME: /usr/local/cuda-11.0
NVCC: Build cuda_11.0_bu.TC445_37.28540450_0
GCC: gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
PyTorch: 1.7.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.2-Product Build 20210312 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.8.0
OpenCV: 4.5.2
MMCV: 1.3.7
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 11.0
MMDetection: 2.13.0+87ad480

**Error traceback**
Traceback (most recent call last):
  File "tools/train.py", line 196, in <module>
    main()
  File "tools/train.py", line 192, in main
    meta=meta)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/mmdet-2.13.0-py3.7.egg/mmdet/apis/train.py", line 170, in train_detector
    runner.run(data_loaders, cfg.workflow)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 125, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 47, in train
    for i, data_batch in enumerate(self.data_loader):
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 435, in __next__
    data = self._next_data()
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 1085, in _next_data
    return self._process_data(data)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 1111, in _process_data
    data.reraise()
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/torch/_utils.py", line 428, in reraise
    raise self.exc_type(msg)
ValueError: Caught ValueError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 198, in _worker_loop
    data = fetcher.fetch(index)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/mmdet-2.13.0-py3.7.egg/mmdet/datasets/dataset_wrappers.py", line 151, in __getitem__
    return self.dataset[idx % self._ori_len]
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/torch/utils/data/dataset.py", line 218, in __getitem__
    return self.datasets[dataset_idx][sample_idx]
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/mmdet-2.13.0-py3.7.egg/mmdet/datasets/custom.py", line 194, in __getitem__
    data = self.prepare_train_img(idx)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/mmdet-2.13.0-py3.7.egg/mmdet/datasets/custom.py", line 217, in prepare_train_img
    return self.pipeline(results)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/mmdet-2.13.0-py3.7.egg/mmdet/datasets/pipelines/compose.py", line 40, in __call__
    data = t(data)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/mmdet-2.13.0-py3.7.egg/mmdet/datasets/pipelines/transforms.py", line 1441, in __call__
    results = self.aug(**results)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/albumentations/core/composition.py", line 188, in __call__
    p.preprocess(data)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/albumentations/core/utils.py", line 62, in preprocess
    data[data_name] = self.check_and_convert(data[data_name], rows, cols, direction="to")
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/albumentations/core/utils.py", line 70, in check_and_convert
    return self.convert_to_albumentations(data, rows, cols)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 51, in convert_to_albumentations
    return convert_bboxes_to_albumentations(data, self.params.format, rows, cols, check_validity=True)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 306, in convert_bboxes_to_albumentations
    return [convert_bbox_to_albumentations(bbox, source_format, rows, cols, check_validity) for bbox in bboxes]
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 306, in <listcomp>
    return [convert_bbox_to_albumentations(bbox, source_format, rows, cols, check_validity) for bbox in bboxes]
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 254, in convert_bbox_to_albumentations
    check_bbox(bbox)
  File "/home/ytx/soft/anaconda3/envs/ipen-mmlab/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 333, in check_bbox
    "to be in the range [0.0, 1.0], got {value}.".format(bbox=bbox, name=name, value=value)
ValueError: Expected x_min for bbox (-0.001953125, 0.18359375, 0.158203125, 0.640625, 0) to be in the range [0.0, 1.0], got -0.001953125.

thank you !

**Comment by Ycrazyprimitive** [comment]
Before I use ’Albu‘ ,  I only use 'RandomFlip' in the original configs, so I can train normally.

**Comment by CharlesNJ** [comment]
What was the issue? @Ycrazyprimitive

**Comment by silverlining21** [comment]
I met the same issue, the ablu config borrowed from `configs/albu_example/mask_rcnn_r50_fpn_albu_1x_coco.py`.

any suggestion will be appericated

**Comment by silverlining21** [comment]
I  try my config without `albu`, which can start training process normaly. I have checked the `ablumenttions` related code, and found during convert `pascal_voc` bbox format to `albumentations` format, here only apply norm on it. 
[https://github.com/albumentations-team/albumentations/blob/4afa308646dd9fa5e7cd37b8dcc38da416a6a5ad/albumentations/augmentations/bbox_utils.py#L257](url)

what's more, I rechecked the voc bbox and guarente the validility  0,0<xmin,ymin<width,hegiht.

so the resaon might lie in  mmdet processing part, I will dig it up.
